### PR TITLE
Remove deprecated DataVolume garbage collection tests

### PR DESCRIPTION
**Pull request by arnongilboa**
### What this PR does
After several releases with CDI `DataVolume` garbage collection disabled by default, we decided to [deprecate](https://github.com/kubevirt/containerized-data-importer/pull/3552) it, as unfortunately it violates fundamental principle of Kubernetes. CR should not be auto-deleted when it completes its role (`Job` with `TTLSecondsAfterFinished` is an exception), and once CR was created we can assume it is there until explicitly deleted. In addition, CR should keep idempotency, so the same CR manifest can be applied multiple times, as long as it is a valid update (e.g. `DataVolume` validation webhook does not allow updating the spec).

### Checklist
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
```release-note
Remove deprecated DataVolume garbage collection tests
```

### CODE REVIEW

File: staging/src/kubevirt.io/api/core/v1/schema.go [hunk 1]
```diff
@@ -857,8 +857,7 @@ type HotplugVolumeSource struct {
 }
 
 type DataVolumeSource struct {
-	// Name of both the DataVolume and the PVC in the same namespace.
-	// After PVC population the DataVolume is garbage collected by default.
+	// Name of the DataVolume in the same namespace.
```
Comment (ShellyKa13) [inline staging/src/kubevirt.io/api/core/v1/schema.go:860]: Data volume source can still be a PVC

**Review by ShellyKa13 — COMMENTED** [review]
One comment, other then that looks good

File: staging/src/kubevirt.io/api/core/v1/schema.go (hunk 1 above)
Comment (arnongilboa) [reply staging/src/kubevirt.io/api/core/v1/schema.go]: Right! Fixed.

File: tests/storage/snapshot.go [hunk 2]
```diff
@@ -1001,6 +1001,54 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 				}
 			})
 
+			It("[test_id:4611] should successfully create a snapshot", func() {
```
Comment (akalenyu) [inline tests/storage/snapshot.go:1004]: Maybe this deserves a separate commit, a bit confusing as to why you'd want to put back a test that was under the GC branch

File: tests/libstorage/datavolume.go [hunk 3]
```diff
@@ -80,36 +77,7 @@ func EventuallyDV(dv *v1beta1.DataVolume, timeoutSec int, matcher gomegatypes.Go
...
-
-	Eventually(func() bool {
-		pvc, err := ThisPVCWith(namespace, name)()
-		Expect(err).ToNot(HaveOccurred())
-		return pvc != nil && pvc.Spec.VolumeName != ""
-	}, timeoutSec, time.Second).Should(BeTrue())
+	Eventually(ThisDVWith(namespace, name), timeoutSec, time.Second).Should(matcher)
 }
 
 func DeleteDataVolume(dv **v1beta1.DataVolume) {
```
Comment (akalenyu) [inline tests/libstorage/datavolume.go:83]: Do we really need this? I can't remember why we had to dive into the **v1beta1.DataVolume realm

**Review by akalenyu — COMMENTED** [review]
Looks good, some question, and
> Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

Should we consider a thread to let (hopefully nobody) know?

File: tests/libstorage/datavolume.go (hunk 3 above)
Comment (arnongilboa) [reply tests/libstorage/datavolume.go:83]: We use `**v1beta1.DataVolume` to set `Context()`-shared dv to nil after we `Delete()` it in the `AfterEach()`, so we won't carry over the dv value of previous test, and we know if it was created/deleted. Removing it will add that same code to some of the `AfterEach`s.

Comment (arnongilboa) [reply tests/libstorage/datavolume.go:83]: Regarding the `EventuallyDV/With()`, I think it's preferable as it's used all over the storage tests and shorter and more readable than the explicit call.

Comment (akalenyu) [reply tests/libstorage/datavolume.go:83]: The DVs are deleted anyway as part of the global `testCleanup` after each test, so we don't have to delete explicitly.
If we have to nil out some global variables, we can do that in the AfterEach blocks which I believe is what we usually do

File: tests/storage/snapshot.go (hunk 2 above)
Comment (akalenyu) [reply tests/storage/snapshot.go:1004]: @ShellyKa13 is there anything special about this test? from what I can tell it's probably something that is exercised already

Comment (ShellyKa13) [reply tests/storage/snapshot.go:1004]: I also thought if this test should be removed, but I saw that it gets into volumes verification with volumesnapshots, which I dont think there is another test that checks that deep.. I figured that it probably existed before the DV GC and that it was converted to also be tested with GC..

Comment (arnongilboa) [reply tests/storage/snapshot.go:1004]: Right, I just reverted the GC split.

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm

**Comment by mhenriks** [comment]
Not sure we should merge this until right before using new CDI release with GC disabled

**Comment by arnongilboa** [comment]
> Not sure we should merge this until right before using new CDI release with GC disabled

Why should we spend many CPU cycles on testing DV garbage collection when it's already disabled by default for several releases, nobody uses it, and we already deprecated it on CDI main?

File: tests/libstorage/datavolume.go (hunk 3 above)
Comment (akalenyu) [reply tests/libstorage/datavolume.go]: Don't need to nil them out? I guess not if the tests pass?

Comment (arnongilboa) [reply tests/libstorage/datavolume.go]: Going over the tests on the last commit, it seems we have no reason to nil them.

Comment (akalenyu) [reply tests/libstorage/datavolume.go]: Great

**Comment by mhenriks** [comment]
>Why should we spend many CPU cycles on testing DV garbage collection when it's already disabled by default for several releases, nobody uses it, and we already deprecated it on CDI main?

I will defer to @akalenyu on this as he surely knows better the effect GC testing is having on CI

**Comment by akalenyu** [comment]
> > Why should we spend many CPU cycles on testing DV garbage collection when it's already disabled by default for several releases, nobody uses it, and we already deprecated it on CDI main?
> 
> I will defer to @akalenyu on this as he surely knows better the effect GC testing is having on CI

If your only worry was CI, then yeah, as soon as the non-GC CDI main comes in, this set of tests is failing consistently. We are removing the tests in this PR, so we should be fine.
/approve

**Comment by akalenyu** [comment]
> > > Why should we spend many CPU cycles on testing DV garbage collection when it's already disabled by default for several releases, nobody uses it, and we already deprecated it on CDI main?
> > 
> > 
> > I will defer to @akalenyu on this as he surely knows better the effect GC testing is having on CI
> 
> If your only worry was CI, then yeah, as soon as the non-GC CDI main comes in, this set of tests is failing consistently. We are removing the tests in this PR, so we should be fine. /approve

I can't approve since the PR is cross sig
